### ISSUE TITLE
Additional suggestions for Proxy IP

### ISSUE THREAD

**Issue by bruhmomentobama** [depthbomb/Scraps#37]
Add a threshold for how long to wait before the Proxy IP gets denied
Because usually, you wait 1-2 seconds for proxies that actually work so a threshold should be useful, as well as adding a "Scan was canceled before the Proxy IP could be fully checked" if no results could be pulled before it got canceled and a "Scan was canceled before 1/3 checks could be completed" if only 1 or 2 or 3, etc. results that couldn't be checked before it reached the time limit to scan it.

Add Proxy Website List Support
So if you're too lazy to manually check Proxy Lists you can pick a preset from one of 3 websites that is available to check proxy lists from and verify which works and which doesn't and whichever proxy works will get added to the list.

Examples being:
https://spys.one/en/http-proxy-list/
https://hidemy.name/en/proxy-list/?type=h#list
http://free-proxy.cz/en/proxylist/country/all/http/ping/all
https://geonode.com/free-proxy-list/ (Check HTTP and HTTPS, then check YES for Google Passed, then check HIA and ANM, thank you @CIusterConsultant for recommending this website!)

Sub addition to the suggestion above: To check proxies only within a set ms threshold, so if you live in the USA and have a couple Proxies from let's say Peru if it goes over for example 200 ms it will skip it and only rely on Proxies minimal to for example 50 ms.

Add a sharing is caring feature.
Basically, if you find working Proxies then you can select to "Share with other ScrapsGUI users" so then they don't need to scan entire websites and can rely on the Proxies that worked out for you.

Add a "Skip temporarily non-working Proxies" option.
Basically, if you have a Proxy IP that passed all the checks but when you start the bot it suddenly had a big wall of red error text telling you the proxy doesn't work, you can temporarily skip it until it becomes valid again.

Edit 1: Thank you @CIusterConsultant for recommending the Proxy IP Site

![Concept](https://user-images.githubusercontent.com/88178558/162602071-be90486c-5712-4030-adec-ede9f2835897.png)

**Comment by bruhmomentobama** [comment]
@depthbomb Pinged cus Idk if you saw this or not, sorry.

**Comment by userbanes** [comment]
i WANT ITTTT

**Comment by thecumship** [comment]
Sounds like a good idea.

**Comment by CIusterConsultant** [comment]
Bumping this cus it's a good idea.

Btw you forgot another proxy list provider, not as popular but eh.
https://geonode.com/free-proxy-list/
Check Http and Https and select yes for google passed then HIA or ANM or both
brings a lot of proxies to use.

Sent here by thecumship

**Comment by userbanes** [comment]
now I'm wondering how this would be implemented in a program scale/perspective also maybe if you don't want your proxy to be shared you can add a tick button also maybe scrap.tf mods can ban the proxies and your account so this doesn't seem too feasible

**Comment by userbanes** [comment]
i feel too many things could go wrong here tbh
(edit: don't get me wrong its a wonderful idea though its kinda yknow)

**Comment by bruhmomentobama** [comment]
> now I'm wondering how this would be implemented in a program scale/perspective also maybe if you don't want your proxy to be shared you can add a tick button also maybe scrap.tf mods can ban the proxies and your account so this doesn't seem too feasible

Yeah but that's why there's a "Share" Button so you can share your proxies.

Also if you look up online some of the proxies are from amazon so most likely banning the proxies are a bad idea.

**Comment by userbanes** [comment]
> > now I'm wondering how this would be implemented in a program scale/perspective also maybe if you don't want your proxy to be shared you can add a tick button also maybe scrap.tf mods can ban the proxies and your account so this doesn't seem too feasible
> 
> Yeah but that's why there's a "Share" Button so you can share your proxies.
> 
> Also if you look up online some of the proxies are from amazon so most likely banning the proxies are a bad idea.

*some*

**Comment by depthbomb** [comment]
The proxy feature will not be expanded upon further than what is already implemented since it would be a lot of work put in to a feature that not many users would (or should) use.

If you are concerned about your IP being seen then you should use a VPN.
